Support more TIFF-based file formats

bokesan opened the pull request:
Many TIFF-based RAW formats work out of the box just by adding their magic numbers. I tested DNG (Leica) and Sony ARW. There is no change needed for DNG since it already matches the first NEF magic numbers.
I did not add test images because of the size and copyright considerations. If this is not acceptable, I could add a Sony ARW test file, but no other formats.

emmanueltouzery commented:
great, thanks for the PR! Yes, I agree, probably no point in gathering more and more huge files in the test suite...

Two things though:

- the test suite breaks:

  tests/Tests.hs:279: 
  1) not a JPG returns empty list if not a JPEG
       doesn't match
       expected: Left "Not a JPEG or NEF file"
        but got: Left "Not a JPEG, TIFF, or TIFF-based raw file"

It's just a matter of changing the expected string, can you do that? you can run the tests with `stack test` or `cabal test`

- maybe you can add a link where you got these IDs from? I have some links at the top of hsexif.hs =>
https://github.com/emmanueltouzery/hsexif/blob/master/Graphics/HsExif.hs#L172

but yeah often these links will be 404s long-term and so on.. but well if you found a good reference, it can't hurt to add the link. if you don't have a good link, just let it be, no big deal.

bokesan commented:
Fixed the failing test.

I don't have a full reference for RAW file magic numbers, but then, they shouldn't be used without some verification anyway. For example, Olympus ORF is also TIFF-based (magic 0x49494F08), but parsing the EXIF data failed, so I removed the magic number again.
